tests/unit/test_opentelemetry_tracing.py

```diff
@@ -0,0 +1,215 @@
+        test_job_ref.created = time_created
+        test_job_ref.started = started_time
+        test_job_ref.ended = ended_time
+        test_job_ref.error_result = error_result
+        test_job_ref.state = "some_job_state"
+
+        with opentelemetry_tracing.create_span(
+            TEST_SPAN_NAME, attributes=TEST_SPAN_ATTRIBUTES, job_ref=test_job_ref
+        ) as span:
+            assert span is not None
+            assert span.name == TEST_SPAN_NAME
+            assert span.attributes == expected_attributes
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_default_no_data_leakage(setup):
+    import google.auth.credentials
+    from google.cloud.bigquery import client
+    from google.cloud.bigquery import job
+
+    mock_credentials = mock.Mock(spec=google.auth.credentials.Credentials)
+    test_client = client.Client(
+        project="test_project", credentials=mock_credentials, location="test_location"
+    )
+
+    expected_attributes = {
+        "foo": "baz",
+        "db.system": "BigQuery",
+        "db.name": "test_project",
+        "location": "test_location",
+    }
+    with opentelemetry_tracing.create_span(
+        TEST_SPAN_NAME, attributes=TEST_SPAN_ATTRIBUTES, client=test_client
+    ) as span:
+        assert span.name == TEST_SPAN_NAME
+        assert span.attributes == expected_attributes
+
+    test_job_reference = job._JobReference(
+        job_id="test_job_id", project="test_project_id", location="test_location"
+    )
+    test_client = client.Client(
+        project="test_project", credentials=mock_credentials, location="test_location"
+    )
+    test_job = job._AsyncJob(job_id=test_job_reference, client=test_client)
+
+    expected_attributes = {
+        "db.system": "BigQuery",
+        "db.name": "test_project_id",
+        "location": "test_location",
+        "num_child_jobs": 0,
+        "job_id": "test_job_id",
+        "foo": "baz",
+        "hasErrors": False,
+    }
+
+    with opentelemetry_tracing.create_span(
+        TEST_SPAN_NAME, attributes=TEST_SPAN_ATTRIBUTES, job_ref=test_job
+    ) as span:
+        assert span.name == TEST_SPAN_NAME
+        assert span.attributes == expected_attributes
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_span_creation_error(setup):
+    import google.auth.credentials
+    from google.cloud.bigquery import client
+    from google.api_core.exceptions import GoogleAPICallError, InvalidArgument
+
+    mock_credentials = mock.Mock(spec=google.auth.credentials.Credentials)
+    test_client = client.Client(
+        project="test_project", credentials=mock_credentials, location="test_location"
+    )
+
+    expected_attributes = {
+        "foo": "baz",
+        "db.system": "BigQuery",
+        "db.name": "test_project",
+        "location": "test_location",
+    }
+    with unittest.TestCase().assertRaises(GoogleAPICallError):
```

Comment:
Could you also check that the status code is set on the span?

Reply:
This test just asserts the exception was thrown there is another test that I wrote that verify's span status. Seen here:
https://github.com/googleapis/python-bigquery/pull/215/files#diff-87020ebeb7f433bac8fc936494fe2c75R1255